src/components/VtkTwoView.vue

```diff
@@ -12,473 +12,305 @@
-
-      // window width
-      this.rangeManipulator.setVerticalListener(
-        0,
-        this.windowing.max - this.windowing.min,
-        1 / 512,
-        () => this.windowing.width,
-        (w) => this.setWindowWidth(w)
-      );
-
-      // window level
-      this.rangeManipulator.setHorizontalListener(
-        this.windowing.min,
-        this.windowing.max,
-        1 / 512,
-        () => this.windowing.level,
-        (l) => this.setWindowLevel(l)
-      );
-
-      // slicing
-      this.rangeManipulator.setScrollListener(
-        this.sliceRange[0],
-        this.sliceRange[1],
-        1,
-        () => this.slice,
-        (s) => this.setSlice(s)
-      );
-    },
-
-    updateOrientationLabels() {
-      const camera = this.view.getCamera();
-      // TODO make modifications only if vup and vdir differ
-      const vup = camera.getViewUp();
-      const vdir = camera.getDirectionOfProjection();
-      const vright = [0, 0, 0];
-      vec3.cross(vright, vdir, vup);
-
-      // assume direction is orthonormal
-      const { direction } = this.worldOrientation;
-
-      // since camera is in "image space", transform into
-      // image's world space.
-      const cameraMat = mat3.fromValues(...vright, ...vup, ...vdir);
-      const imageMat = mat3.fromValues(...direction);
-      // `direction` is row-major, and gl-matrix is col-major
-      mat3.transpose(imageMat, imageMat);
-      const cameraInImWorld = mat3.create();
-      mat3.mul(cameraInImWorld, imageMat, cameraMat);
-
-      // gl-matrix is col-major
-      const left = cameraInImWorld.slice(0, 3).map((c) => -c);
-      const up = cameraInImWorld.slice(3, 6);
-
-      let leftLabels = lpsDirToLabels(left);
-      let upLabels = lpsDirToLabels(up);
-
-      // sort by magnitude
-      leftLabels = zip(left.map(Math.abs), leftLabels)
-        .sort(([a], [b]) => b - a)
-        .map(([, label]) => label)
-        .join('');
-      upLabels = zip(up.map(Math.abs), upLabels)
-        .sort(([a], [b]) => b - a)
-        .map(([, label]) => label)
-        .join('');
-
-      this.view.setCornerAnnotation('n', upLabels);
-      this.view.setCornerAnnotation('w', leftLabels);
-    },
-
-    setupInteraction() {
-      const istyle = this.view.getInteractorStyle2D();
-
-      // removes all manipulators
-      InteractionPresets.applyDefinitions(
-        [
-          { type: 'pan', options: { shift: true } },
-          { type: 'zoom', options: { control: true } },
-          { type: 'zoom', options: { button: 3 } },
-        ],
-        istyle
-      );
-
-      // create our own set of manipulators
-      istyle.addMouseManipulator(this.rangeManipulator);
-    },
-
-    setWindowWidth(width) {
-      if (this.baseImageExists) {
-        this.setWindowing({ width });
-      }
-    },
+    // setup ranges for mouse controls
+    const wwRange = computed(() => ({
+      min: 0,
+      max: windowing.value.max - windowing.value.min,
+      step: 1 / 512,
```

Comment:
This was an arbitrary step value. I'm not actually sure what step value we should be using for window/level changes, and quite honestly this probably should be computed off of the pixel data range of the image.

Reply:
I'm going to mark this as resolved for now, but I'll open a new issue. I'm not sure how dynamic this should be, or if this is a safe constant. This will be something we can address laer.